### PR TITLE
Add tests for handling of NULL datum in CDefaultComparator in ORCA

### PR DESCRIPTION
**Pull request by sambitesh**

### CODE REVIEW

File: src/test/regress/sql/gporca.sql [hunk 1]
```diff
@@ -1476,6 +1476,12 @@ CREATE CAST (myint AS numeric) WITH FUNCTION myint_numeric(myint) AS IMPLICIT;
 EXPLAIN SELECT a FROM csq_cast_param_outer WHERE b in (SELECT CASE WHEN a > 1 THEN d ELSE '42' END FROM csq_cast_param_inner);
 SELECT a FROM csq_cast_param_outer WHERE b in (SELECT CASE WHEN a > 1 THEN d ELSE '42' END FROM csq_cast_param_inner);
 
+-- Query should not fall back to planner when "IN"/"NOT IN" clause contains NULL for char/text datatypes
+
+EXPLAIN SELECT a FROM ggg WHERE a NOT IN (NULL, '');
+
+EXPLAIN SELECT a FROM ggg WHERE a IN (NULL, 'x');
```
Comment (bchaudhary) [inline src/test/regress/sql/gporca.sql]: add a tests to validate the results

File: src/test/regress/expected/gporca.out [hunk 2]
```diff
@@ -10170,3 +10170,103 @@ SELECT a FROM csq_cast_param_outer WHERE b in (SELECT CASE WHEN a > 1 THEN d ELS
...
+                                 QUERY PLAN                                 
+----------------------------------------------------------------------------
+ Gather Motion 3:1  (slice1; segments: 3)  (cost=0.00..1.01 rows=1 width=2)
+   ->  Seq Scan on ggg  (cost=0.00..1.01 rows=1 width=2)
+         Filter: a = ANY ('{NULL,x}'::bpchar[])
+ Optimizer: legacy query optimizer
+(4 rows)
+
+-- start_ignore
+DROP SCHEMA orca CASCADE;
```
Comment (dhanashreek89) [inline src/test/regress/expected/gporca.out]: Don't need to copy the ignore block output in expected file
